buildstockbatch/schemas/v0.3.yaml

```diff
@@ -19,11 +19,18 @@ postprocessing: include('postprocessing-spec', required=False)
 references: map(required=False)
 ---
 gcp-spec:
-  job_identifier: regex('^[a-zA-Z]\w{,9}$', required=True)
+  job_identifier: regex('^[a-zA-Z_-]\w{,20}$', required=True)
+  gcp_project: str(required=True)
```

Comment:
why is this prefixed with 'gcp_' but the others not? I actually agree the clarity is good, but wonder if we can be more consistent — could it make sense to prefix the other things that are GCP parameters, like region and artifact_registry?